docs/reference/services/project.rst

```diff
@@ -60,6 +62,9 @@ At this step, any application-specific transformations of the document are appli
 using the :py:meth:`ProjectService._app_preprocess_project` method. By default,
 nothing happens here.
 
+The project dict at this state is available using the
+:py:meth:`ProjectService._preprocess` protected method.
```

Comment:
I find this is confusing because `_app_preprocess_project` is called by `_preprocess`.
If I knew nothing about the internals of the lib I would think I need to call `_preprocess` in `_app_preprocess_project`.

Reply:
That's a good point. What do you think of this language:

```rst
The project dict at this state is available using the
:py:meth:`ProjectService._preprocess` protected method. ``_preprocess`` calls
``_app_preprocess_project`` as the last step of processing before returning the
pre-processed project dict.
```